core/src/main/scala/codesearch/core/search/Search.scala

```diff
@@ -95,25 +113,27 @@ trait Searcher {
     }
   }
 
-  protected def runCsearch(arg: SearchArguments): IO[List[String]] = {
-    val pathRegex     = if (arg.sourcesOnly) langExts else ".*"
-    val query: String = if (arg.preciseMatch) Helper.hideSymbols(arg.query) else arg.query
-
-    val args: mutable.ListBuffer[String] = mutable.ListBuffer("csearch", "-n")
-    if (arg.insensitive) {
-      args.append("-i")
-    }
-    args.append("-f", pathRegex)
-    args.append(query)
-
-    IO {
-      (Process(args, None, "CSEARCHINDEX" -> indexPath.toString()) #| Seq("head", "-1001")).!!.split('\n').toList
+  /**
+    * Returns index of first matched lines and lines of code
+    */
+  private def extractRows(path: String, codeLine: Int, beforeLines: Int, afterLines: Int): IO[(Int, Seq[String])] = {
+    readFileAsync(path).map { lines =>
+      val (code, indexes) = lines.zipWithIndex.filter {
+        case (_, index) => index >= codeLine - beforeLines - 1 && index <= codeLine + afterLines
+      }.unzip
+      indexes.head -> code
     }
   }
 
 }
 
-object Searcher {
+object Search {
+
+  private[search] val snippetConfig: SnippetConfig =
+    Config.load
+      .map(_.snippetConfig)
+      .handleErrorWith(_ => IO(SnippetConfig(pageSize = 30, linesBefore = 3, linesAfter = 5)))
```

Comment:
Let if fail. We want to know if config is wrong.

Reply:
Ok.